R/fit_model.R

```diff
@@ -0,0 +1,174 @@
+#' Fit an EpiNow2 model
+#'
+#' @param data As returned from [CFAEpiNow2Pipeline::read_data()]
+#' @param parameters As returned from
+#'   [CFAEpiNow2Pipeline::read_disease_parameters()]
+#' @param seed The random seed, used for both initialization by EpiNow2 in R and
+#'   sampling in Stan
+#' @param horizon The number of days, as an integer, to forecast
+#' @param priors A list of lists. The first level should contain the key `rt`
+#'   with elements `mean` and `sd` and the key `gp` with element `alpha_sd`.
+#' @param sampler_opts A list. The Stan sampler options to be passed through
+#'   EpiNow2. It has required keys: `cores`, `chains`, `iter_warmup`,
+#'   `iter_sampling`, `max_treedepth`, and `adapt_delta`.
+#'
+#' @return A fitted model object of class `epinow` or, if model fitting fails,
+#'   an NA is returned with a warning
+#' @export
+fit_model <- function(
+    data,
+    parameters,
+    seed,
+    horizon,
+    priors,
+    sampler_opts) {
+  # Priors ------------------------------------------------------------------
+  rt <- EpiNow2::rt_opts(
+    list(
+      mean = priors[["rt"]][["mean"]],
+      sd = priors[["rt"]][["sd"]]
+    )
+  )
+  gp <- EpiNow2::gp_opts(
+    alpha_sd = priors[["gp"]][["alpha_sd"]]
+  )
+
+  # Distributions -----------------------------------------------------------
```

Comment:
Not in scope for this PR, but I think we need to do a better job documenting and explaining our choice to always discretize the distribution ourselves and pass the pmf

Reply:
#29